src/main/java/cash/z/ecc/android/sdk/tool/SaplingParamTool.kt

```diff
@@ -0,0 +1,135 @@
+package cash.z.ecc.android.sdk.tool
+
+import cash.z.ecc.android.sdk.exception.TransactionEncoderException
+import cash.z.ecc.android.sdk.ext.Bush
+import cash.z.ecc.android.sdk.ext.ZcashSdk
+import cash.z.ecc.android.sdk.ext.twig
+import cash.z.ecc.android.sdk.ext.twigTask
+import com.squareup.okhttp.OkHttpClient
+import com.squareup.okhttp.Request
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okio.Okio
+import java.io.File
+
+class SaplingParamTool {
+
+    companion object {
+        /**
+         * Checks the given directory for the output and spending params and calls [fetchParams] if
+         * they're missing.
+         *
+         * @param destinationDir the directory where the params should be stored.
+         */
+        suspend fun ensureParams(destinationDir: String) {
+            var hadError = false
+            arrayOf(
+                ZcashSdk.SPEND_PARAM_FILE_NAME,
+                ZcashSdk.OUTPUT_PARAM_FILE_NAME
+            ).forEach { paramFileName ->
+                if (!File(destinationDir, paramFileName).exists()) {
```

Comment:
Good question @pacu . Previously, the directory we were passing to these functions was controlled by the SDK. Now that this tool exposes that behavior we should do something to ensure that the developer is passing the correct directory (rather than putting param files in an area that is not private to the app, which is a [potential security issue](https://github.com/zcash/zcash-android-wallet-sdk/issues/30))

We should add some protections that ensure the developer is using the right directory but I think that maybe falls outside the scope of this PR. I'd like to punt and give it some thought and maybe run it by Taylor and then immediately apply a change after this is merged.